src/lib/prov/tpm2/tpm2_crypto_backend/tpm2_crypto_backend.cpp

```diff
@@ -566,16 +573,66 @@ static TSS2_RC get_ecdh_point(TPM2B_PUBLIC* key,
                               BYTE* out_buffer,
                               size_t* out_size,
                               void* userdata) {
+   #if defined(BOTAN_HAS_TPM2_ECC_ADAPTER)
+   return thunk([&] {
+      BOTAN_UNUSED(userdata);
+      BOTAN_ASSERT_NONNULL(key);
+      BOTAN_ASSERT_NOMSG(key->publicArea.type == TPM2_ALG_ECC);
+
+      const auto curve_name = Botan::TPM2::curve_id_tss2_to_botan(key->publicArea.parameters.eccDetail.curveID);
```

Comment:
Perhaps it is worth considering to let `curve_id_tss2_to_botan()` perform the transformation into an `EC_Group` as well. I believe in the ECC code we also just transform the curve name into an `EC_Group` right away.

Reply:
We can actually get the `EC_Group` if we load the TPM EC PK first - exactly the same operation is performed there. Then, there's no redundant transformation code.
(Additionally, that way all methods in `tpm2_algo_mappings.h` consistently remain mappings between algo strings/ids which makes me slightly happier than introducing methods that create instances of the algos).